### ISSUE TITLE
Static Key

### ISSUE THREAD

**Issue by aaronbarnardsound** [jb55/lnsocket#8]
Is it possible to have a static public key so that the remote node can limit the rune to that key? Or does a new key need to be created every time it is initialised with the `gen_key` method?

**Comment by jb55** [comment]
yes this will be easy to add and why there is a generate_key step. I planned on having a set_key function as well. I haven't had the need for it yet though which is why it doesn't exist.

**Comment by aaronbarnardsound** [comment]
Yeah the use case I am thinking is for a server to communicate to a core ln node. The idea I am playing with would be a simple API that can be deployed to run in front of your node at home. So it would be great to have a rune with full permissions, but limit it to the public key of the lnsocket.

**Comment by jb55** [comment]
On Mon, Jul 11, 2022 at 03:52:09PM -0700, Aaron Barnard wrote:
>Yeah the use case I am thinking is for a server to communicate to a core ln node. The idea I am playing with would be a simple API that can be deployed to run in front of your node at home. So it would be great to have a rune with full permissions, but limit it to the public key of the lnsocket.

kk I'll add this soon

**Comment by aaronbarnardsound** [comment]
Legend thank you!

**Comment by jb55** [comment]
there is now an `lnsocket_setkey` in v0.3.2 for setting static keys. Let me know if you run into any issues!

**Comment by jb55** [comment]
hmm it might be good to have a method that gets the pubkey...

**Comment by aaronbarnardsound** [comment]
Awesome will give this a go.

**Comment by aaronbarnardsound** [comment]
Yeah a method that gets the pubkey would be super handy. Saves having to load `tiny-secp256k1` in the browser to derive the pub key.

**Comment by aaronbarnardsound** [comment]
Or even a method to get the keypair after generating the key. That way you can use set key in a later session with the private key

**Comment by jb55** [comment]
yup it's definitely needed... will add in a bit
